### PR TITLE
feature/firebase-config

### LINKED ISSUE #86: Add firebase config to exportable app
### Feature Description

Allow the cookie cutter to inject firebase config to the exportable app.
Once the app has access to the firebase config, inject the config into the network layer to connect the network layer to firebase/firestore.

### Use Case

Different deployments have different firebase config. The app should take the config as a prop.

### Solution

_Please describe your ideal solution._

### Alternatives

_Please describe any alternatives you've considered, even if you've dismissed them._

### Screenshots/References

_In the case of a visually oriented feature or component request, providing drawings, screenshots, and/or links to examples of what you're seeking will be very helpful._

### PR DESCRIPTION
**Pull request by tohuynh**
### Link to Relevant Issue

This pull request resolves #86 

### Description of Changes

Add a React context(a way to pass data through the component tree without manually passing down props) to contain the firebase config (and possibly other configurations in the future). For us, they are global static data such as firebase config, information about the deployment such as the municipal's name (for use on the home page), and maybe even styling themes.

Then any component that subscribes to this React context, can use a hook `useAppConfigContext` to use the context values. A component can subscribe to the React context by being a descendent of the `AppConfigProvider` component.

In cookie cutter it will look something like this:
`<AppConfigProvider appConfig={ actual app config}><App /><AppConfigProvider />`

In the PR, I added `useAppConfigContext` to get the firebase configs in the `EventPage` and `PersonPage` and pass them to the network layer.

I also copied over the data fetching architecture from the `EventPage` to the `PersonContainer` and `PersonPage`. However, the data fetched in the `PersonPage` is currently insufficient to render the person container. That's saved for #60.

### Link to Forked Storybook Site

_If component changes (especially visual changes) are contained in this PR, we ask that you provide a link to a publicly viewable version of the Storybook docs site, so PR reviewers can see your changes without having to install and view your code locally._

_Please see `CONTRIBUTING.md` for directions on how this can be done._

### CODE REVIEW

File: src/pages/PersonPage/PersonPage.tsx [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+        if (!didCancel) {
+          personDataDispatch({
+            type: FetchDataActionType.FETCH_SUCCESS,
+            payload: person,
+          });
+        }
+      } catch (err) {
+        if (!didCancel) {
+          const error = createError(err);
+          personDataDispatch({ type: FetchDataActionType.FETCH_FAILTURE, payload: error });
```
Comment (JacksonMaxfield) [inline src/pages/PersonPage/PersonPage.tsx:48]: I think I just found a typo:

`FETCH_FAILTURE` -> `FETCH_FAILURE`

**Review by JacksonMaxfield — APPROVED** [review]
This literally all looks perfect besides for the typo I found.

Amazing.

**Review by BrianL3 — APPROVED** [review]
Thanks To.  I was midway through the PersonContainer stuff with the new pattern but this is fine, speeds things up really. 

Passing the config through each Page is not ideal IMO, but I can see why its set up this way and have no better idea.  Optimally, I would think, only the NetworkService class would actually have any idea about the Firestore context to keep it completely decoupled from the view layers.  But it seems difficult to get it there.  Can useAppConfigContext not be imported into NetworkServices directly (as they are not React components?)?

**Comment by tohuynh** [comment]
> Can useAppConfigContext not be imported into NetworkServices directly (as they are not React components?)?
Yeah, useAppConfigContext is React hook and can be used only in React functional components.

**Pull request merged**